syft/pkg/cataloger/cataloger.go

```diff
@@ -80,5 +80,28 @@ func AllCatalogers(cfg Config) []Cataloger {
 		golang.NewGoModuleBinaryCataloger(),
 		golang.NewGoModFileCataloger(),
 		rust.NewCargoLockCataloger(),
+	}, cfg.EnabledCatalogers)
+}
+
+func filterCatalogers(catalogers []Cataloger, enabledCatalogers []string) []Cataloger {
+	// if enable-cataloger is not set, all applicable catalogers are enabled by default
+	if len(enabledCatalogers) == 0 {
+		return catalogers
+	}
+	var filteredCatalogers []Cataloger
+	for _, cataloger := range catalogers {
```

Comment:
we could add additional logic such that the string `-cataloger` is optional on each user-specified name, to be nicer to the user.